src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

```diff
@@ -313,9 +313,10 @@ private void processFiles(List<File> files) throws CheckstyleException {
     private SortedSet<LocalizedMessage> processFile(File file) throws CheckstyleException {
         final SortedSet<LocalizedMessage> fileMessages = new TreeSet<>();
         try {
-            final FileText theText = new FileText(file.getAbsoluteFile(), charset);
+            final File canonicalFile = file.getCanonicalFile();
+            final FileText theText = new FileText(canonicalFile, charset);
             for (final FileSetCheck fsc : fileSetChecks) {
-                fileMessages.addAll(fsc.process(file, theText));
+                fileMessages.addAll(fsc.process(canonicalFile, theText));
```

Comment:
By changes like this you potentially make problems too all other Checks (do not forget about thousands of third-party Checks).

https://stackoverflow.com/a/1099370/1015848

> Converting from a path to a canonical path makes it absolute (usually tack on the current working directory so e.g. ./file.txt becomes c:/temp/file.txt). The canonical path of a file just "purifies" the path, removing and resolving stuff like ..\ and resolving symlinks (on unixes).

canonical is purified absolute path. Why we should do this ? it is not Check business. Some path given to Checker and it should just propagate it to Checks with minimal changes (user always right).

Fix should be in Check only.

Reply:
Done. Made fix for current Check only.